fix(client): persist transcript expanded state

moT01 opened the pull request:
This expands the transcripts by default - and persists the state in local storage if the user changes it.
Also changes the heading text to "Expand / Collapse Transcript" and makes it bold - to help users know it's clickable. 

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Ref: [58418](https://github.com/freeCodeCamp/freeCodeCamp/issues/58418)

<!-- Feel free to add any additional description of changes below this line -->

moT01 committed:
fix: persist transcript expanded state

Dario-DC commented on client/i18n/locales/english/translations.json:
```diff
@@ -429,6 +429,8 @@
     "answered-mcq": "You have unanswered questions and/or incorrect answers.",
     "explanation": "Explanation",
     "transcript": "Read a transcript of this video",
```
Is this still needed?

moT01 replied:
No, it's not. For some reason I kinda like to keep them around in case we want to use it again, but there's really no need to.

moT01 committed:
fix: remove unused i18n key

jdwilkin4 commented on client/src/templates/Challenges/components/challenge-transcript.tsx:
```diff
@@ -14,11 +15,28 @@ function ChallengeTranscript({
 
+  // default to expanded
+  const [isOpen, setIsOpen] = useState(
+    () => (store.get('fcc-transcript-expanded') as boolean | null) ?? true
```
What's the consensus for this?

moT01 replied:
Seems like most people don't want them open by default.

jdwilkin4 replied:
so, doesn't this need to be changed to false?

```suggestion
    () => (store.get('fcc-transcript-expanded') as boolean | null) ?? false
```

moT01 replied:
yes, or something like that. Most of the people that weighed in did like them collapsed by default, but Quincy said he wanted them open by default. So I'm not sure what to do - I was going to ask more of the team and see what they think and maybe run it by Quincy again if that's the consensus. So for now, we can just block this.

moT01 committed:
fix: default to collapsed + text

moT01 commented:
Alright, I think we can go with it like this. I reverted the default state to collapsed and adjusted the text to just "Transcript", like described in the issue. If we want to change the default, we can on another PR - but this seems to be the consensus.

ojeytonwilliams reviewed (approved):
LGTM :+1:

moT01 committed:
chore: add tests

moT01 committed:
fix: comment

moT01 committed:
fix: use test id for e2e selector

ojeytonwilliams commented on e2e/challenge-transcript.spec.ts:
Tests are good, but could we just unit test this? It's just an isolated component - it doesn't even talk to redux.

moT01 replied:
✅

moT01 committed:
fix: tests again

moT01 committed:
fix: better

ojeytonwilliams commented:
Thanks, Tom. That looks good.

ShaunSHamilton reviewed (approved):
LGTM 👍
